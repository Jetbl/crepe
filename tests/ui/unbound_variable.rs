// In this test, the variable named 'n' is used before binding in a clause.

mod datalog {
    use crepe::crepe;

    crepe! {
        @output
        struct Fib(u32, u32);

        Fib(0, 0);
        Fib(1, 1);

        Fib(n, x + y) <- Fib(n - 1, x), Fib(n - 2, y), (n <= 25);
    }
}

fn main() {}
