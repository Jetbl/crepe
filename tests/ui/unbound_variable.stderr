error[E0425]: cannot find value `n` in this scope
  --> tests/ui/unbound_variable.rs:13:30
   |
13 |         Fib(n, x + y) <- Fib(n - 1, x), Fib(n - 2, y), (n <= 25);
   |                              ^ not found in this scope

error[E0425]: cannot find value `n` in this scope
  --> tests/ui/unbound_variable.rs:13:45
   |
13 |         Fib(n, x + y) <- Fib(n - 1, x), Fib(n - 2, y), (n <= 25);
   |                                             ^ help: a local variable with a similar name exists: `x`

error[E0425]: cannot find value `n` in this scope
  --> tests/ui/unbound_variable.rs:13:57
   |
13 |         Fib(n, x + y) <- Fib(n - 1, x), Fib(n - 2, y), (n <= 25);
   |                                                         ^ help: a local variable with a similar name exists: `x`

error[E0425]: cannot find value `n` in this scope
  --> tests/ui/unbound_variable.rs:13:13
   |
13 |         Fib(n, x + y) <- Fib(n - 1, x), Fib(n - 2, y), (n <= 25);
   |             ^ help: a local variable with a similar name exists: `x`
